### PR TITLE
use iframe to sandbox generated html

### LINKED ISSUE #1294: xss vulnerability in demo page
**Describe the bug**

Our [demo](https://marked.js.org/demo/?text=scenario%201%20-%20%3Cjavascript%3Aalert(1)%3E%0A%0A%5Bscenario%202%5D(javascript%3Aalert(2))%0A%0A%5Bscenario%203%5D%5B1%5D%0A%0A%5Bscenario%204%5D%0A%0A%5B1%5D%3A%20javascript%3Aalert(3)%0A%0A%5BScenario%204%5D%3A%20javascript%3Aalert(4)) page allows xss

We should try to mitigate this similar to commonmark's [demo](https://spec.commonmark.org/dingus/?text=scenario%201%20-%20%3Cjavascript%3Aalert(1)%3E%0A%0A%5Bscenario%202%5D(javascript%3Aalert(2))%0A%0A%5Bscenario%203%5D%5B1%5D%0A%0A%5Bscenario%204%5D%0A%0A%5B1%5D%3A%20javascript%3Aalert(3)%0A%0A%5BScenario%204%5D%3A%20javascript%3Aalert(4)) page

We should also have a disclaimer in our documentation about possible xss vulnerability with displaying unsanitized output

related #1232

**To Reproduce**

1. goto the [demo](https://marked.js.org/demo/?text=scenario%201%20-%20%3Cjavascript%3Aalert(1)%3E%0A%0A%5Bscenario%202%5D(javascript%3Aalert(2))%0A%0A%5Bscenario%203%5D%5B1%5D%0A%0A%5Bscenario%204%5D%0A%0A%5B1%5D%3A%20javascript%3Aalert(3)%0A%0A%5BScenario%204%5D%3A%20javascript%3Aalert(4))
2. click one of the links

**Expected behavior**
Javascript should not execute

### PR DESCRIPTION
**Pull request by UziTech**
## Description

- Use an iframe to prevent xss on demo page
- Adds a warning to the readme and docs for displaying unsanitized output HTML
- Fixes #1294

## Contributor

- [ ] Test(s) exist to ensure functionality and minimize regression (if no tests added, list tests covering this PR); or,
- [X] no tests required for this PR.
- [ ] If submitting new feature, it has been documented in the appropriate places.

## Committer

In most cases, this should be a different person than the contributor.

- [ ] Draft GitHub release notes have been updated.
- [ ] CI is green (no forced merge required).
- [ ] Merge PR

### CODE REVIEW

**Comment by styfle** [comment]
Isn’t there a sanitize option in marked to prevent XSS? Or did we already remove that?

**Comment by styfle** [comment]
Ok if that's the case, should we add this iframe solution to the [USING_ADVANCED.md](https://marked.js.org/#/USING_ADVANCED.md) docs to make it the preferred method of sanitizing rather than the `sanitize` option?

**Comment by UziTech** [comment]
The iframe solution wouldn't work well in most situations.

I think we should recommend something like DOMPurify if they are displaying user generated content.

```
DOMPurify.sanitize(marked(...));
```

**Comment by styfle** [comment]
> The iframe solution wouldn't work well in most situations.

Why use it here? Maybe the demo should use DOMPurify?

**Comment by UziTech** [comment]
Because it's a demo. I don't want the user to see sanitized output if they don't specify the sanitize option

**Comment by styfle** [comment]
@UziTech That makes sense.

------

@joshbruce or @davisjam can you take a look at this?

**Comment by lirantal** [comment]
Thanks for the security disclaimer notice, appreciate the awareness you set for users 👍

File: README.md [hunk 1]
```diff
@@ -29,7 +29,9 @@ Our [documentation pages](https://marked.js.org) are also rendered using marked
 
 **In-browser:** `npm install marked --save`
 
-## Usage 
+## Usage
+
+### Warning: 🚨 Marked does not sanitize the output HTML by default 🚨
```
Comment (davisjam) [inline README.md]: Could link to the docs on `sanitize` here.

**Comment by davisjam** [comment]
I don't see the `sanitize` option documented clearly anywhere. The `USING_ADVANCED.md` docs say:

```
|sanitize    |`boolean` |`false`  |???      |If true, sanitize the HTML passed into `markdownString` with the `sanitizer` function.|
```

This sounds a bit circular to me.

Thoughts on:
- Enhancing the description there (and explicitly mentioning the XSS possibility)?
- Directing readers to the advanced docs from the regular docs for security considerations?

**Comment by UziTech** [comment]
I still say we deprecate the `sanitize` and `sanitizer` options and recommend

```js
DOMPurify.sanitize(marked(...));
```

like in https://github.com/markedjs/marked/issues/1232#issuecomment-382603925

**Comment by styfle** [comment]
I'm going to merge this because it addresses sanitization of marked.js.org and we shouldn't leave this XSS vulnerability lingering any longer.

If anyone wants to update the docs to make this sanitation process clearer, feel free to submit a new PR.

Thanks!

**Pull request merged**